### PR TITLE
UI: Provide option to hide default toolbar tools

### PR DESCRIPTION
**Pull request by Tomastomaslol**
Issue: #14617 

## What I did

* Added ids to all preview tools
* Added new filter to preview `toolbar` that will exclude any and toolbar item that has matching been excluded with configuration or story parameter
* added `toolbar` config to addons `Config` that to be able to hide any toolbar item.

### Story parameter scenarios

* added new stories  for `preview.stories.tsx` for following scenarios:
* * `Hide fullscreen`
* * `Hide all default tools`

### Configuration Scenarios

```js
addons.setConfig({
  toolbar: { fullscreen: { hidden: true } },
});
```

Results in:

Screen Shot 2021-05-12 at 11 22 18 am

```js
addons.setConfig({
  toolbar: {
    title: { hidden: true },
    zoom: { hidden: true },
    eject: { hidden: true },
    copy: { hidden: true },
    fullscreen: { hidden: true },
    'storybook/background': { hidden: true },
    'storybook/viewport': { hidden: true },},
});
```

Results in:
Screen Shot 2021-05-12 at 11 28 33 am

### CODE REVIEW

File: lib/ui/src/components/preview/toolbar.tsx [hunk 1]
```diff
@@ -165,6 +167,16 @@ export const Tools = React.memo<{ list: Addon[] }>(({ list }) => (
...
 ));
 
+function toolbarItemHasBeenExcluded(item: Partial<Addon>, story: PreviewProps['story']) {
+  const toolbarItemsFromStoryParameters =
+    'toolbar' in story.parameters ? story.parameters.toolbar : undefined;
+  const { toolbar: toolbarItemsFromAddonsConfig } = addons.getConfig();
+
+  const toolbarItems = merge(toolbarItemsFromAddonsConfig, toolbarItemsFromStoryParameters);
+
+  return toolbarItems ? !!toolbarItems[item.id]?.hidden : false;
```
Comment (Tomastomaslol) [inline lib/ui/src/components/preview/toolbar.tsx:177]: is ok  to use `?` ?

**Comment by Tomastomaslol** [comment]
Before this PR can be considered is there any documentation I need to update or add?

File: lib/ui/src/components/preview/toolbar.tsx (hunk 1 above)
Comment (shilman) [reply lib/ui/src/components/preview/toolbar.tsx:177]: sure! 👍

**Comment by shilman** [comment]
@Tomastomaslol probably here? https://storybook.js.org/docs/react/configure/features-and-behavior#gatsby-focus-wrapper

**Comment by Tomastomaslol** [comment]
@shilman Thanks for the pointer! I added some documentation for the feature 🙂

File: docs/configure/features-and-behavior.md [hunk 2]
```diff
@@ -36,3 +37,9 @@ The following options are configurable under the `sidebar` namespace:
 | **showRoots**         | Boolean       |Display the top-level nodes as a "root" in the sidebar         |`false`                                         |
 | **collapsedRoots**    | Array         |Set of root node IDs to visually collapse by default           |`['misc', 'other']`                             |
 | **renderLabel**       | Function      |Create a custom label for tree nodes; must return a ReactNode  |`(item) => <abbr title="...">{item.name}</abbr>`|
+
+The following options are configurable under the `toolbar` namespace:
+
+| Name                  | Type          | Description                                                   | Example Value                                  |
+| ----------------------|:-------------:|:-------------------------------------------------------------:|:----------------------------------------------:|
+| **id**                | String        |Toggle visibility for toolbar item                             |` { hidden: false }`                            |
```
Comment (shilman) [inline docs/configure/features-and-behavior.md:45]:
```suggestion
| **id**                | String        |Toggle visibility for toolbar item                             |`{ hidden: false }`                            |
```

**Review by shilman — APPROVED** [review]
Great fix @Tomastomaslol 💯

**Comment by ndelangen** [comment]
We should extract these default tools into separate addons at some point

**Pull request merged**